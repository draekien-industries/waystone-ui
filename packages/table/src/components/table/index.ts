export * from './table.body';
export { TableContainer } from './table.container';
export * from './table.head';
