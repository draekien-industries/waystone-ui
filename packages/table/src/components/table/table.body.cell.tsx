import { flexRender, type Cell } from '@tanstack/react-table';

export const TableBodyCell = <D, V>({ id, column, getContext }: Cell<D, V>) => {
  const { columnDef, getSize } = column;

  return (
    <td
      id={id}
      sx={{
        width: getSize(),
        paddingY: 'small',
        paddingX: 'medium',
      }}
    >
      {flexRender(columnDef.cell, getContext())}
    </td>
  );
};

const VirtualizedTableBodyCell = <D, V>({
  id,
  column,
  getContext,
}: Cell<D, V>) => {
  const { columnDef, getSize } = column;

  return (
    <td
      id={id}
      sx={{
        display: 'flex',
        width: getSize(),
        paddingY: 'small',
        paddingX: 'medium',
      }}
    >
      {flexRender(columnDef.cell, getContext())}
    </td>
  );
};

TableBodyCell.Virtualized = VirtualizedTableBodyCell;
