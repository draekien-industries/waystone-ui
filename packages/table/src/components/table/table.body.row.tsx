import type { Row } from '@tanstack/react-table';
import { alpha } from '@theme-ui/color';
import { TableBodyCell } from './table.body.cell';

export const TableBodyRow = <D,>({
  id,
  getCanSelect,
  getVisibleCells,
}: Row<D>) => {
  const enabled = getCanSelect();
  const cells = getVisibleCells();

  return (
    <tr
      id={id}
      sx={{
        color: enabled ? 'text' : 'muted',
        borderBottom: '1px solid',
        borderBottomColor: alpha('muted', 0.25),
      }}
    >
      {cells.map((cell) => (
        <TableBodyCell key={cell.id} {...cell} />
      ))}
    </tr>
  );
};
