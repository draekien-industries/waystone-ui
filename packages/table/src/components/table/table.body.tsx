import { alpha } from '@theme-ui/color';
import type { PropsWithChildren } from 'react';
import { TableBodyRow } from './table.body.row';
import { TableBodyCell } from './table.body.cell';

export const TableBody = ({ children }: PropsWithChildren) => (
  <tbody
    sx={{
      'tr:nth-of-type(even)': {
        backgroundColor: alpha('b-400', 0.1),
      },
    }}
  >
    {children}
  </tbody>
);

TableBody.Row = TableBodyRow;
TableBody.Cell = TableBodyCell;
