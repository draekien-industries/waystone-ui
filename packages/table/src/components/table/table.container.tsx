import type { PropsWithChildren } from 'react';
import type { ThemeUICSSObject } from 'theme-ui';

export const style = {
  width: '100%',
  minWidth: '400px',
  boxShadow: 'medium',
  borderCollapse: 'collapse',
} as const satisfies ThemeUICSSObject;

export const TableContainer = ({ children }: PropsWithChildren) => (
  <table sx={style}>{children}</table>
);
