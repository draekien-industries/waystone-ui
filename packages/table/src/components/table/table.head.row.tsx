import type { PropsWithChildren } from 'react';
import type { HeaderGroup as TableHeadRowProps } from '@tanstack/react-table';
import { TableHeadCell } from './table.head.cell';

export const TableHeadRow = <D,>({ id, headers }: TableHeadRowProps<D>) => (
  <tr id={id}>
    {headers.map((header) => (
      <TableHeadCell key={header.id} {...header} />
    ))}
  </tr>
);
