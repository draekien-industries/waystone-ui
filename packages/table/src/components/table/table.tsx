import type { PropsWithChildren } from 'react';

export const WaystoneTable = ({ children }: PropsWithChildren) => (
  <table>{children}</table>
);
