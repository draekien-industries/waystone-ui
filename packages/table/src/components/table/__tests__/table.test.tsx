/* eslint-disable react/jsx-props-no-spreading */
import {
  getCoreRowModel,
  getExpandedRowModel,
  useReactTable,
  type ExpandedState,
} from '@tanstack/react-table';
import { render, screen, userEvent } from '@waystone/testing-library';
import { useState } from 'react';
import { columns, data } from '../../../__fixtures__';
import { TableBody } from '../table.body';
import { TableContainer } from '../table.container';
import { TableHead } from '../table.head';

function ExampleTable() {
  const [expanded, setExpanded] = useState<ExpandedState>({});

  const { getHeaderGroups, getRowModel } = useReactTable({
    data,
    columns,
    state: {
      expanded,
    },
    onExpandedChange: setExpanded,
    getSubRows: (row) => row.subRows,
    getCoreRowModel: getCoreRowModel(),
    getExpandedRowModel: getExpandedRowModel(),
    debugTable: true,
  });

  return (
    <TableContainer>
      <TableHead>
        {getHeaderGroups().map((group) => (
          <TableHead.Row key={group.id} {...group} />
        ))}
      </TableHead>
      <TableBody>
        {getRowModel().rows.map((row) => (
          <TableBody.Row key={row.id} {...row} />
        ))}
      </TableBody>
    </TableContainer>
  );
}

describe('<Table />', () => {
  it('should match snapshot', () => {
    const { container } = render(<ExampleTable />);

    expect(container).toMatchSnapshot();
  });

  it('should expand the table when the expand button in the header is clicked', async () => {
    const user = userEvent.setup({ delay: null });

    render(<ExampleTable />);

    const expandAllButton = screen.getAllByText('expand_more')[0];

    await user.click(expandAllButton);

    const expandAllButtons = screen.queryAllByText('expand_more');
    const collapseAllButtons = await screen.findAllByText('expand_less');

    expect(expandAllButtons).toHaveLength(0);
    expect(collapseAllButtons.length).toBeGreaterThan(0);
  });
});
