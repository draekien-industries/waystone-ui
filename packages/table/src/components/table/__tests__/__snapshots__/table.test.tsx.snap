// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match snapshot 1`] = `
<div>
  <table
    class="css-1ks362w-TableContainer"
  >
    <thead
      class="css-1kuoydi-TableHead"
    >
      <tr
        id="0"
      >
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="table-interactions"
        >
          <div
            class="css-g89m8z-Box"
          >
            <div
              class="css-v65un1-InteractiveColumnHeader"
            >
              <div
                class="css-17w1b6b-CheckboxWithRef"
              >
                <input
                  class="css-1pckfla-CheckboxWithRef"
                  id="select-all-checkbox"
                  type="checkbox"
                />
              </div>
              <button
                class="css-1e7uob3-Box"
              >
                <i
                  class="material-icons-outlined css-1ye7pjf-Icon"
                  name="expand_more"
                >
                  expand_more
                </i>
              </button>
            </div>
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="firstName"
        >
          <div
            class="css-g89m8z-Box"
          >
            First Name
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="lastName"
        >
          <div
            class="css-g89m8z-Box"
          >
            Last Name
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="age"
        >
          <div
            class="css-g89m8z-Box"
          >
            Age
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="visits"
        >
          <div
            class="css-g89m8z-Box"
          >
            Visits
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="status"
        >
          <div
            class="css-g89m8z-Box"
          >
            Status
          </div>
        </th>
        <th
          class="css-ifvbdr-TableHeadCell"
          colspan="1"
          id="progress"
        >
          <div
            class="css-g89m8z-Box"
          >
            Progress
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="css-1om1htk-TableBody"
    >
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="0"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="0_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_0"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_firstName"
        >
          Mike
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_lastName"
        >
          Rowe
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_age"
        >
          3
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_visits"
        >
          705
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="0_progress"
        >
          72
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="1"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="1_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_1"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_firstName"
        >
          Palma
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_lastName"
        >
          Bergnaum
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_age"
        >
          37
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_visits"
        >
          522
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="1_progress"
        >
          11
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="2"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="2_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_2"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_firstName"
        >
          Reinhold
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_lastName"
        >
          Bernhard
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_age"
        >
          30
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_visits"
        >
          305
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="2_progress"
        >
          76
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="3"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="3_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_3"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_firstName"
        >
          Vernice
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_lastName"
        >
          Barton
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_age"
        >
          31
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_visits"
        >
          892
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="3_progress"
        >
          8
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="4"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="4_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_4"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_firstName"
        >
          Mandy
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_lastName"
        >
          Kertzmann
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_age"
        >
          30
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_visits"
        >
          390
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="4_progress"
        >
          66
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="5"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="5_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_5"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_firstName"
        >
          Justen
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_lastName"
        >
          Runolfsdottir
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_age"
        >
          40
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_visits"
        >
          571
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="5_progress"
        >
          30
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="6"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="6_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_6"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_firstName"
        >
          Ashleigh
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_lastName"
        >
          Cremin
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_age"
        >
          4
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_visits"
        >
          257
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="6_progress"
        >
          97
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="7"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="7_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_7"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_firstName"
        >
          Annetta
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_lastName"
        >
          Graham-Bernhard
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_age"
        >
          40
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_visits"
        >
          494
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_status"
        >
          complicated
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="7_progress"
        >
          1
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="8"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="8_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_8"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_firstName"
        >
          Monte
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_lastName"
        >
          Bayer
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_age"
        >
          27
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_visits"
        >
          363
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="8_progress"
        >
          9
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="9"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="9_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_9"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_firstName"
        >
          Deonte
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_lastName"
        >
          McGlynn
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_age"
        >
          35
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_visits"
        >
          211
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_status"
        >
          complicated
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="9_progress"
        >
          34
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="10"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="10_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_10"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_firstName"
        >
          Bryce
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_lastName"
        >
          Hagenes
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_age"
        >
          13
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_visits"
        >
          288
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="10_progress"
        >
          89
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="11"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="11_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_11"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_firstName"
        >
          Kamren
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_lastName"
        >
          Murphy
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_age"
        >
          9
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_visits"
        >
          447
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="11_progress"
        >
          24
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="12"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="12_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_12"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_firstName"
        >
          Betty
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_lastName"
        >
          Hessel
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_age"
        >
          28
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_visits"
        >
          626
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="12_progress"
        >
          67
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="13"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="13_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_13"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_firstName"
        >
          Lou
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_lastName"
        >
          Jakubowski
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_age"
        >
          40
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_visits"
        >
          73
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="13_progress"
        >
          99
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="14"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="14_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_14"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_firstName"
        >
          Lempi
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_lastName"
        >
          Wyman
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_age"
        >
          8
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_visits"
        >
          420
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="14_progress"
        >
          34
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="15"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="15_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_15"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_firstName"
        >
          Jade
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_lastName"
        >
          Baumbach
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_age"
        >
          7
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_visits"
        >
          367
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="15_progress"
        >
          21
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="16"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="16_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_16"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_firstName"
        >
          Cleta
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_lastName"
        >
          McLaughlin
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_age"
        >
          38
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_visits"
        >
          379
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="16_progress"
        >
          11
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="17"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="17_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_17"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_firstName"
        >
          Stan
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_lastName"
        >
          Heathcote
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_age"
        >
          30
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_visits"
        >
          725
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_status"
        >
          relationship
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="17_progress"
        >
          16
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="18"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="18_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_18"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_firstName"
        >
          Ozella
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_lastName"
        >
          Deckow
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_age"
        >
          31
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_visits"
        >
          942
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_status"
        >
          complicated
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="18_progress"
        >
          21
        </td>
      </tr>
      <tr
        class="css-1n4gb8y-TableBodyRow"
        id="19"
      >
        <td
          class="css-vvureu-TableBodyCell"
          id="19_table-interactions"
        >
          <div
            class="css-1dxymne-InteractiveColumnCell"
          >
            <div
              class="css-17w1b6b-CheckboxWithRef"
            >
              <input
                class="css-1pckfla-CheckboxWithRef"
                id="checkbox_19"
                type="checkbox"
              />
            </div>
            <button
              class="css-1e7uob3-Box"
            >
              <i
                class="material-icons-outlined css-1ye7pjf-Icon"
                name="expand_more"
              >
                expand_more
              </i>
            </button>
          </div>
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_firstName"
        >
          Bobbie
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_lastName"
        >
          Harber
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_age"
        >
          1
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_visits"
        >
          161
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_status"
        >
          single
        </td>
        <td
          class="css-vvureu-TableBodyCell"
          id="19_progress"
        >
          3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
